openedx/tests/xblock_integration/test_review_xblock.py

```diff
@@ -0,0 +1,529 @@
+                category='problem',
+                display_name='Problem 3'
+            )
+            cls.xblock4_review = ItemFactory.create(
+                parent=cls.unit1_review,
+                category='problem',
+                display_name='Problem 4'
+            )
+            cls.unit2_review = ItemFactory.create(
+                parent=cls.section_review, display_name='New Unit 2'
+            )
+            cls.xblock5_review = ItemFactory.create(
+                parent=cls.unit2_review,
+                category='problem',
+                display_name='Problem 5'
+            )
+            cls.unit3_review = ItemFactory.create(
+                parent=cls.section_review, display_name='New Unit 3'
+            )
+            cls.xblock6_review = ItemFactory.create(
+                parent=cls.unit3_review,
+                category='problem',
+                display_name='Problem 6'
+            )
+
+        cls.course_review_url = reverse(
+            'courseware_section',
+            kwargs={
+                'course_id': unicode(cls.course_review.id),
+                'chapter': 'Overview',
+                'section': 'Welcome',
+            }
+        )
+
+    def setUp(self):
+        super(TestReviewXBlock, self).setUp()
+
+        for idx, student in enumerate(self.STUDENTS):
+            username = 'u{}'.format(idx)
+            self.create_account(username, student['email'], student['password'])
+            self.activate_user(student['email'])
+
+        self.staff_user = GlobalStaffFactory()
+
+    def enroll_student(self, email, password, course):
+        """
+        Student login and enroll for the course
+        """
+        self.login(email, password)
+        self.enroll(course, verify=True)
+
+
+@attr(shard=1)
+@ddt.ddt
+class TestReviewFunctions(TestReviewXBlock):
+    """
+    Check that the essential functions of the Review xBlock work as expected.
+    Tests cover the basic process of receiving a hint, adding a new hint,
+    and rating/reporting hints.
+    """
+    def test_no_review_problems(self):
+        """
+        If a user has not seen any problems, they should
+        receive a response to go out and try more problems so they have
+        material to review.
+        """
+        self.enroll_student(self.STUDENTS[0]['email'], self.STUDENTS[0]['password'], self.course_actual)
+        self.enroll_student(self.STUDENTS[0]['email'], self.STUDENTS[0]['password'], self.course_review)
+
+        with self.store.bulk_operations(self.course_actual.id, emit_signals=False):
+            review_section_actual = ItemFactory.create(
+                parent=self.chapter_actual, display_name='Review Subsection'
+            )
+            review_unit_actual = ItemFactory.create(
+                parent=review_section_actual, display_name='Review Unit'
+            )
+
+            review_xblock_actual = ItemFactory.create(  # pylint: disable=unused-variable
```

Comment:
I'm curious how you handled creating these blocks multiple times without them messing up the results in other tests.

Reply:
If you technically look at the course, under that chapter, you will see many Review Subsections since we are making a new one per test. But the test knows to look at the subsection created within it because the response is loading that local variable.

Reply:
👍